components/inputs/input-time.js

```diff
@@ -80,24 +112,33 @@ class InputTime extends LitElement {
 	constructor() {
 		super();
 		this.disabled = false;
+		this.enforceTimeIntervals = false;
+		this.timeInterval = 'thirty';
 		this.labelHidden = false;
-		this._value = formatValue(DEFAULT_VALUE);
+		this._dropdownId = getUniqueId();
 		this._formattedValue = formatTime(DEFAULT_VALUE);
 		this._timezone = formatTime(new Date(), {format: 'ZZZ'});
-		this._dropdownId = getUniqueId();
+		this._value = formatValue(DEFAULT_VALUE);
 	}
 
 	get value() { return this._value; }
 	set value(val) {
 		const oldValue = this.value;
 		const time = parseValue(val);
+		if (this.enforceTimeIntervals) {
+			const interval = getIntervalNumber(this.timeInterval);
+			const difference = time.getMinutes() % interval;
+			if (difference > 0) {
+				time.setMinutes(time.getMinutes() + interval - difference);
+			}
+		}
 		this._value = formatValue(time);
 		this._formattedValue = formatTime(time);
 		this.requestUpdate('value', oldValue);
 	}
 
 	render() {
-		getIntervals();
+		getIntervals(this.timeInterval);
```

Comment:
If it's not going to actually return anything, better to call this `initIntervals()` or something.